src/libsyntax/parse/parser.rs

```diff
@@ -1486,6 +1486,15 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
+    /// Parses `extern` followed by an optional ABI string, or nothing.
+    fn parse_extern_abi(&mut self) -> PResult<'a, Abi> {
```

Comment:
Btw, I was going over `parser/ty.rs` and I noticed that `parse_ty_bare_fn` has the exact same logic. Would be good to de-duplicate that.